core/game_object.lua

```diff
@@ -2419,165 +2423,237 @@ function loadAPIs()
-                G.SETTINGS.selected_colours[self.type] = G.SETTINGS.selected_colours[self.type] or
-                SMODS.Palettes[self.type]["Default"]
-            end
-            if SMODS.Palettes[self.type][self.name] then
-                G.FUNCS.update_atlas(self.type)
-                return
+            self.name = (self.name:len() < 18 and self.name or self.name:sub(1,16).."...")
+            -- Do not create palettes for any types that do not exist
+            if not G.P_CENTER_POOLS[self.type] and self.type ~= "Suit" and self.type ~= "Blind" or SMODS.AltTextures[self.type] and SMODS.AltTextures[self.type][self.name] then return end
+            
+            -- Initialize new palette types, including a default palette 
+            if not SMODS.AltTextures[self.type] then
+                table.insert(SMODS.AltTextures.Types, self.type)
+                SMODS.AltTextures[self.type] = {names = {}}
+            end
+            if #SMODS.AltTextures[self.type].names == 0 then
+                if self.name ~= "Default" then SMODS.AltTexture:create_default(self.type) end
             end
-            table.insert(SMODS.Palettes[self.type].names, self.name)
-            SMODS.Palettes[self.type][self.name] = {
+            table.insert(SMODS.AltTextures[self.type].names, self.name)
+
+            SMODS.AltTextures[self.type][self.name] = {
                 name = self.name,
-                order = #SMODS.Palettes[self.type].names,
-                old_colours = {},
-                new_colours = {}
+                order = #SMODS.AltTextures[self.type].names,
             }
-            if self.old_colours then
-                for i = 1, #self.old_colours do
-                    SMODS.Palettes[self.type][self.name].old_colours[i] = type(self.old_colours[i]) == "string" and
-                    HEX(self.old_colours[i]) or self.old_colours[i]
-                    SMODS.Palettes[self.type][self.name].new_colours[i] = type(self.new_colours[i]) == "string" and
-                    HEX(self.new_colours[i]) or self.new_colours[i]
-                end
-            end
-            if not G.SETTINGS.selected_colours[self.type] then
-                G.SETTINGS.selected_colours[self.type] = SMODS.Palettes[self.type][self.name]
+            if self.type == "Suit" then
+                SMODS.AltTextures[self.type][self.name].suits = self.suits or SMODS.AltTextures[self.type]["Default"].suits
             end
 
-            SMODS.Palette:create_atlas(self.type, self.name)
-            G.FUNCS.update_atlas(self.type)
+            -- Initialize the new palette
+            if self.palette then self:create_palette()
+            elseif self.texture then self:create_new_atlas() end
         end
     }
 
-    function SMODS.Palette:create_default(type)
-        table.insert(SMODS.Palettes[type].names, "Default")
-        SMODS.Palettes[type]["Default"] = {
+    function SMODS.AltTexture:create_palette()
```

Comment:
from_palette, from_atlas seem more natural. after all, we aren't creating a palette, we're creating a texture from a palette